daemon/codec.c

```diff
@@ -1717,7 +1717,14 @@ void codec_add_raw_packet(struct media_packet *mp, unsigned int clockrate) {
 #ifdef WITH_TRANSCODING
 static void codec_add_raw_packet_dup(struct media_packet *mp, unsigned int clockrate) {
 	struct codec_packet *p = g_slice_alloc0(sizeof(*p));
-	str_init_dup_str(&p->s, &mp->raw);
+	// don't just duplicate the string. need to ensure enough room
+	// if encryption is enabled on this stream
+	char *buf = g_malloc(mp->raw.len + 1 + RTP_BUFFER_TAIL_ROOM);
+	if (mp->raw.s && mp->raw.len)
+		memcpy(buf, mp->raw.s, mp->raw.len);
+	buf[mp->raw.len] = '\0';
```

Comment:
Good catch, thanks. I'll see if I can add a test to catch this error. There's no need for the null termination here though as this isn't a C string.

Reply:
oki, i tried to stay close to the functionality of `str_init_dup_str` that this is replacing so kept it in. will remove it though